src/components/NucleotideEntropyMultiChart.tsx

```diff
@@ -0,0 +1,294 @@
+                as='select'
+                value={gene}
+                onChange={ev => setGene(ev.target.value)}
+                className='flex-grow'
+                size='sm'
+              >
+                {jsonRefData.genes.map(g => (
+                  <option value={g.name} key={g?.name}>
+                    {g.name}
+                  </option>
+                ))}
+              </Form.Control>
+            </div>
+          }
+        </div>
+    );
+
+
+
+    let plotArea;
+    if (variants![0].length == 0) {
+      plotArea = <p>Select one or more lineages to compare.</p>
+    } else if (!data) {
+      plotArea = <Loader />;
+    } else {
+        plotArea = (
+        <Plot
+            plotData={data.dayArray}
+            ticks={data.ticks}
+            variants={variants!}
+            sequenceType={sequenceType}
+        />
+        );
+    }
+    
+    return (
+        <>
+        {controls}
+        {plotArea}
+        </>
+      );
+}
+
+type PlotProps = {
+    plotData: TransformedTime;
+    ticks: string[];
+    variants: string[];
+    sequenceType: SequenceType;
+};
+
+const useData = (
+    selectors: LapisSelector[],
+    selectedGene: GeneOption[],
+    variants: any,
+    sequenceType: SequenceType,
+    deletions: boolean
+): any | undefined => {
+  //fetch the proportions per position in weekly segments
+  const weeklyVariantMutationProportionQuery = useQuery(
+    async signal => {
+      //calculate weeks
+      const dayArray: UnifiedDay[] = [selectors[0].dateRange?.getDateRange().dateFrom!, selectors[0].dateRange?.getDateRange().dateTo!];
+      const dayRange = globalDateCache.rangeFromDays(dayArray)!;
+      const weeks = globalDateCache.weeksFromRange({ min: dayRange.min.isoWeek, max: dayRange.max.isoWeek });
+      const weekDateRanges = new Array<DateRange>();
+      for (let i = 0; i < weeks.length; i++) {
+        let dateRange: DateRange = { dateFrom: weeks[i].firstDay, dateTo: weeks[i].firstDay };
+        weekDateRanges[i] = dateRange;
+      }
+
+      const weekSelectors = new Array<LapisSelector>(); 
+      selectors.forEach(selector => {
+          for (let w of weekDateRanges) { 
+            weekSelectors.push({
+            ...selector,
+            dateRange: new FixedDateRangeSelector(w),
+          })
+        };
+      })
+
+      let weekLength = weekDateRanges.length;
+
+      return {
+        sequenceType: sequenceType,
+        weekNumber: weekLength,
+        result: await Promise.all(
+          weekSelectors.map((w, i) =>
+            MutationProportionData.fromApi(w, sequenceType, signal).then(data => {
+              const proportions: MutationProportionEntry[] = data.payload.map(m => m);
+              let date = weekDateRanges[i % weekLength];
+              return {
+                proportions,
+                date,
+              };
+            })
+          )
+        )
+      }
+    },
+    [selectors, sequenceType]
+  );
+    
+  const data = useMemo(() => {
```

Comment:
This could be inlined:
```suggestion
  return useMemo(() => {
```